lib/commons/color/get-background-color.js

```diff
@@ -54,7 +54,8 @@ function _getBackgroundColor(elm, bgElms, shadowOutlineEmMax) {
   }
 
   const textRects = getVisibleChildTextRects(elm);
-  let bgColors = getTextShadowColors(elm, { minRatio: shadowOutlineEmMax });
+  let bgColors =
+    getTextShadowColors(elm, { minRatio: shadowOutlineEmMax }) ?? [];
```

Comment:
For the `?? []` to come up in `color-contrast` and not have already resulted in a `complexTextShadows` exit in `color-contrast-evaluate`, an element would need to have a "complex" (2-3 edge compound) shadow of blur radius between .001em and 0.2em. I think if that were to actually happen in practice *and* there was additionally a higher-blur-radius shadow(s), it probably doesn't make sense to silently throw away those higher-blur-radius shadow(s). I can be persuaded that either of "bail as incomplete" or "ignore those cases but still consider the thick shadow behind as a possible background color" would be reasonable.

`link-in-text-block` also uses `getBackgroundColor` and I think the `?? []` is even sketchier there, since it doesn't early-exit with incomplete for blur radius < .001 complex compound shadows like `color-contrast` does. Generally, it seems a bit strange to me that `link-in-text-block-evaluate` isn't reusing the same support for shadows providing contrast that color contrast is.

Reply:
I can't sneak anything past you can I? I realised that too when I wrote this. Wasn't sure whether I should care, but fair enough.